tests/ui/lifetimes/mismatched-elided-lifetime-styles.rs

```diff
@@ -0,0 +1,279 @@
+#![deny(mismatched_elided_lifetime_styles)]
+
+//
+//
+
+#[derive(Copy, Clone)]
+struct ContainsLifetime<'a>(&'a u8);
+
+struct S(u8);
+
+fn hidden_ref_to_anonymous_ref(v: &u8) -> &'_ u8 {
+    //~^ ERROR lifetime flowing from input to output with different syntax
+    v
+}
+
+fn anonymous_ref_to_hidden_ref(v: &'_ u8) -> &u8 {
+    //~^ ERROR lifetime flowing from input to output with different syntax
+    v
+}
```

Comment:
These cases should not be linted against.

We agreed that `&u8` is an elided lifetime, not a hidden one.  These are both therefore in the same category, so they are not mismatched.